### DISCUSSION TITLE
frameit 'Infinite' error while ceating frames around screenshots

### DISCUSSION THREAD

**Discussion by damiancioo** [fastlane/fastlane#19970] [Q&A] [answered]
Hi, 
I'm trying to use frameit to create frames around screenshots. 
I'm using  MacOS Monterey 12.1 with M1

My line:

```
  lane :makeFrames do |options|
    frame_screenshots(
      path: "./fastlane/screenshots",
      use_platform: "ANDROID",
      force_device_type: "Nexus 5X",
      debug_mode: true
    )
  end
```

My Framefile.json:
```
{
  "default": {
    "keyword": {
      "font": "./fonts/Roboto-Regular.ttf"
    },
    "title": {
      "font": "./fonts/Roboto-Regular.ttf",
      "color": "#545454"
    },
    "background": "./background.jpg",
    "padding": 50,
    "offset": {
      "offset": "+676+479",
      "titleHeight": 320
    }
  },
  "data": [
    {
      "filter": "screen1",
      "keyword": {
        "color": "#d21559"
      },
      "title": {
        "text": "Hello!"
      }
    }
  ]
}
```

Output:
```
ping-android % bundle exec fastlane makeFrames
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
[✔] 🚀 
+----------------------------+---------+------------+
|                   Used plugins                    |
+----------------------------+---------+------------+
| Plugin                     | Version | Action     |
+----------------------------+---------+------------+
| fastlane-plugin-bundletool | 1.0.4   | bundletool |
+----------------------------+---------+------------+

objc[9002]: Class AMSupportURLConnectionDelegate is implemented in both /usr/lib/libauthinstall.dylib (0x212f6eb90) and /Library/Apple/System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x1039e02c8). One of the two will be used. Which one is undefined.
objc[9002]: Class AMSupportURLSession is implemented in both /usr/lib/libauthinstall.dylib (0x212f6ebe0) and /Library/Apple/System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x1039e0318). One of the two will be used. Which one is undefined.
[12:56:59]: ------------------------------
[12:56:59]: --- Step: default_platform ---
[12:56:59]: ------------------------------
[12:56:59]: Driving the lane 'android makeFrames' 🚀
[12:56:59]: -------------------------------
[12:56:59]: --- Step: frame_screenshots ---
[12:56:59]: -------------------------------
[12:56:59]: Framing screenshots at path ./fastlane/screenshots (via frameit)
[12:56:59]: Using device frames version 'latest'
[⠸] Framing screenshot './pl-PL/screen1.png' [12:57:00]: Infinity

+------+-------------------+-------------+
|            fastlane summary            |
+------+-------------------+-------------+
| Step | Action            | Time (in s) |
+------+-------------------+-------------+
| 1    | default_platform  | 0           |
| 2    | frame_screenshots | 1           |
+------+-------------------+-------------+

[12:57:00]: fastlane.tools finished successfully 🎉
```

Frames are not generated, I dont know what this 'Infinite' red text means :/
Any ideas?  

Env:

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                          |
| --------------------------- | ---------------------------------------------- |
| OS                          | 12.1                                           |
| Ruby                        | 3.1.0                                          |
| Bundler?                    | true                                           |
| Git                         | git version 2.32.0 (Apple Git-132)             |
| Installation Source         | /opt/homebrew/lib/ruby/gems/3.1.0/bin/fastlane |
| Host                        | macOS 12.1 (21C52)                             |
| Ruby Lib Dir                | /opt/homebrew/Cellar/ruby/3.1.0/lib            |
| OpenSSL Version             | OpenSSL 1.1.1m  14 Dec 2021                    |
| Is contained                | false                                          |
| Is homebrew                 | false                                          |
| Is installed via Fabric.app | false                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/    |
| Xcode Version               | 13.2.1                                         |
| Swift Version               | 5.5.2                                          |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

# const
BUNDLETOOL_VERSION = '1.8.2' # https://github.com/google/bundletool/releases

default_platform(:android)

platform :android do


  desc "Generate screenshots frames with "
  desc "INFO: not working yet"
  lane :makeFrames do |options|

    # docs:
    # https://docs.fastlane.tools/actions/frameit/
    frame_screenshots(
      path: screenshotsPath,
      use_platform: "ANDROID",
      force_device_type: "Nexus 5X",
      debug_mode: false
    )
  end

  end

end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
json_key_file("fastlane-service-account.json") # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
package_name("com.h4lsoft.ping") # e.g. com.krausefx.app

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.204.3 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                     | Version | Update-Status |
| -------------------------- | ------- | ------------- |
| fastlane-plugin-bundletool | 1.0.4   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| error_highlight                 | 0.3.0        |
| did_you_mean                    | 1.6.1        |
| bundler                         | 2.1.4        |
| pathname                        | 0.2.0        |
| set                             | 1.0.2        |
| tsort                           | 0.1.0        |
| rake                            | 13.0.6       |
| rexml                           | 3.2.5        |
| CFPropertyList                  | 3.0.5        |
| public_suffix                   | 4.0.6        |
| addressable                     | 2.8.0        |
| artifactory                     | 3.0.15       |
| atomos                          | 0.1.3        |
| aws-eventstream                 | 1.2.0        |
| aws-partitions                  | 1.554.0      |
| aws-sigv4                       | 1.4.0        |
| jmespath                        | 1.5.0        |
| aws-sdk-core                    | 3.126.0      |
| aws-sdk-kms                     | 1.54.0       |
| aws-sdk-s3                      | 1.112.0      |
| babosa                          | 1.0.4        |
| claide                          | 1.1.0        |
| colored                         | 1.2          |
| colored2                        | 3.1.2        |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| declarative                     | 0.0.20       |
| digest-crc                      | 0.6.4        |
| unf_ext                         | 0.0.8        |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| dotenv                          | 2.7.6        |
| emoji_regex                     | 3.2.3        |
| excon                           | 0.91.0       |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| multipart-post                  | 2.0.0        |
| faraday-multipart               | 1.0.3        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| faraday-retry                   | 1.0.3        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.9.3        |
| http-cookie                     | 1.0.4        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.2.0        |
| fastimage                       | 2.2.6        |
| gh_inspector                    | 1.1.3        |
| jwt                             | 2.3.0        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.4        |
| signet                          | 0.16.0       |
| googleauth                      | 1.1.0        |
| httpclient                      | 2.8.3        |
| mini_mime                       | 1.1.2        |
| trailblazer-option              | 0.1.2        |
| uber                            | 0.1.0        |
| representable                   | 3.1.1        |
| retriable                       | 3.1.2        |
| webrick                         | 1.7.0        |
| google-apis-core                | 0.4.2        |
| google-apis-androidpublisher_v3 | 0.16.0       |
| google-apis-playcustomapp_v1    | 0.7.0        |
| google-apis-iamcredentials_v1   | 0.10.0       |
| google-apis-storage_v1          | 0.11.0       |
| google-cloud-env                | 1.5.0        |
| google-cloud-errors             | 1.2.0        |
| google-cloud-core               | 1.6.0        |
| google-cloud-storage            | 1.36.1       |
| json                            | 2.6.1        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| optparse                        | 0.1.1        |
| plist                           | 3.6.0        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.3        |
| simctl                          | 1.6.8        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.8.0        |
| terminal-table                  | 1.8.0        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| word_wrap                       | 1.0.0        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.21.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| xcpretty-travis-formatter       | 1.0.1        |
| fastlane-plugin-bundletool      | 1.0.4        |
</details>

*generated on:* **2022-02-14**
</details>

**Comment by damiancioo** [comment, marked as answer]
Ok, I managed to get rid off this message be removing `offset` property.

**Comment by hekuli** [comment]
Having the same problem when I try to use offset.
